[Cases] Add incremental id service and expose the ID in the UI

janmonschke opened the pull request:
## Summary

This adds and enables the case id incrementer service ([design doc](https://docs.google.com/document/d/1DZKTPl7UryYjpjVMNhIYbE82OADVOg93-d02f0ZQtUI/edit?tab=t.0#heading=h.6qjc4qynaeuo)). In order not to stress bulk creation of cases, we're processing incremental ids asynchronously, meaning they will not immediately appear in the UI.

The feature is currently disabled by default to allow for testing in additional environments after merging but can be enabled by setting `xpack.cases.incrementalIdService.enabled=true` in `kibana(.dev).yml`. Once the flag is enabled, actually rendering the IDs in the UI is disabled by default (for now) and has to be enabled in the advanced settings (`cases:incrementalIdDisplay:enabled`). 

Cases can be found by their incremental ID by searching for `#{incremental_case_id}` in the cases table.

### Screenshots

**Incremental ID in the case detail page**

<img width="1506" alt="Screenshot 2025-06-05 at 15 46 42" src="https://github.com/user-attachments/assets/f51ae0cd-a2e8-48f7-a6db-05f9f1285e95" />

**Incremental ID in the cases table**

<img width="1240" alt="Screenshot 2025-06-05 at 20 32 32" src="https://github.com/user-attachments/assets/619b3f12-1986-4bc7-b9e8-f7556d0c546c" />

**Searching for case by its incremental ID**
<img width="1239" alt="Screenshot 2025-06-05 at 20 33 36" src="https://github.com/user-attachments/assets/771df512-7436-4aa0-88f9-ac3e1e161455" />

### Testing notes

<details>
<summary>Validation script</summary>

Use this script to investigate if there are duplicates or gaps:

```js
import * as fs from 'fs';

// Query to get all cases from all namespaces sorted by incremental_id
// GET .kibana_alerting_cases/_search?_source_excludes=*
// {
//     "query": {
//         "exists": {
//             "field": "cases.incremental_id"
//         }
//     },
//     "fields": [
//       "cases.incremental_id",
//       "cases.title",
//       "namespaces"
//     ],
//     "from": 0,
//     "size": 10000,
//     "sort": [
//       {
//         "cases.incremental_id": {
//           "order": "asc"
//         }
//       }
//     ]
// }
// Put those results into `test.json` in the same directory

// You might need to add `"search_after": [40007]` in case you want to look at more than 10k cases.
// In that case, replace `[40007]` with whatever value the last item has in `"sort": [2102]`

// Concatenate hits if needed (10k per file)
const cases = [
  JSON.parse(fs.readFileSync('./test.json')),
  // JSON.parse(fs.readFileSync('./test1.json')),
  // JSON.parse(fs.readFileSync('./test2.json')),
  // JSON.parse(fs.readFileSync('./test3.json')),
  // JSON.parse(fs.readFileSync('./test4.json')),
].reduce((allHits, currResult) => {
  return allHits.concat(currResult.hits.hits);
}, []);

console.log(`Total amount of cases: ${cases.length}`);

// Groups cases but
const casesByNamespace = cases.reduce((acc, theCase) => {
  const id = theCase._id;
  const space = theCase.fields.namespaces[0];
  const incrementalId = theCase.fields['cases.incremental_id'][0];
  const title = theCase.fields['cases.title'][0];
  const toStore = { id, incrementalId, title };
  if (!acc[space]) {
    acc[space] = new Map();
  }

  // check for duplicates
  const spaceMap = acc[space];
  if (!spaceMap.has(incrementalId)) {
    acc[space].set(incrementalId, toStore);
  } else {
    const storedCase = spaceMap.get(incrementalId);
    console.error(`
      ${storedCase.title} and ${toStore.title} have the same incremental id (${incrementalId})
    `);
  }
  return acc;
}, {});

// find gaps in spaces
Object.keys(casesByNamespace).forEach((space) => {
  const spaceHits = casesByNamespace[space];
  const gaps = [];
  spaceHits.forEach(({ incrementalId }, _, map) => {
    const idBefore = incrementalId - 1;
    if (incrementalId > 1 && !map.has(idBefore)) {
      gaps.push(idBefore);
    }
  });

  console.log(`space:${space} has ${spaceHits.size} cases and ${gaps.length} skipped ids`);
  gaps.forEach((gap) => console.log(`id #${gap} is not assigned`));
});

```


</details>

- Enable the logger in your `kibana.dev.yml` (optional but helpful)
```
logging.loggers:
  - name: plugins.cases.incremental_id_task
    level: debug
```
- Change some of the timings in `x-pack/platform/plugins/shared/cases/server/tasks/incremental_id/incremental_id_task_manager.ts`
  - Set `timeout: '1m'`
  - Set `CASES_INCREMENTAL_ID_SYNC_INTERVAL_DEFAULT_MINUTES = 1`
  - Remove ```runAt: new Date(
            new Date().getTime() + CASES_INCREMENTAL_ID_SYNC_INTERVAL_DEFAULT_MINUTES * 60 * 1000
          ),```
  - you can also set the timings to something lower in the seconds e.g. `10s`
- Generate a bunch of cases with the generator script `x-pack/platform/plugins/shared/cases/scripts/generate_cases.js`: 
  - `node scripts/generate_cases.js -c 1000 -o securitySolution
- Enable `cases:incrementalIdDisplay:enabled` in advanced settings
- Wait a couple minutes until the incrementer task ran
- Test that the ids show up and that the search works

### Research notes

- We ran a large-scale test with ~350k cases in a cloud env and can report the following findings:
  - The 10min timeout for the incremental id task makes sense. The task was usually finished after around 8-9min (processing 1000 cases at a time) which gives it some buffer even.
  - While processing the first 50k cases, the service skipped 8 ids and no duplicates have been assigned. This means it skipped `0.016%` ids which is great.
  - It's unclear when these skips happened though and we investigated the first 50k cases for duplicate ids, just in case, and found no duplicates.
  - At no point did any of the error logs trigger, meaning the task is running smoothly.

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

janmonschke committed:
add initial ID incrementer service

janmonschke committed:
add incremental id to table and case detail page

janmonschke committed:
allow to search by incremental_id

janmonschke committed:
tests++

janmonschke committed:
remove enzyme while I'm here

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/306043)
* Commit: 174e27d8fdb2e108f4b2529a88650f8d10cecceb

### Failed CI Steps
* [Linting (with types)](https://buildkite.com/elastic/kibana-pull-request/builds/306043#01974415-1cfa-4921-ba99-a2924cc0d662)

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306043","number":306043,"commit":"174e27d8fdb2e108f4b2529a88650f8d10cecceb"}],"number":306043}
buildkite-pr-comment-->

janmonschke committed:
promises

janmonschke committed:
void promises

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/306065)
* Commit: 9e72e8ca566e03e0a53c85f37fec4561319f0902

### Failed CI Steps
* [FTR Configs #14](https://buildkite.com/elastic/kibana-pull-request/builds/306065#0197446e-293b-487c-b060-72b28074b750)
* [FTR Configs #128](https://buildkite.com/elastic/kibana-pull-request/builds/306065#0197446e-2971-41ba-9442-48557ebb66ab)
* [FTR Configs #128](https://buildkite.com/elastic/kibana-pull-request/builds/306065#01974479-34ec-4927-a198-0ce7d4849d0a)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/306065#0197446e-293b-487c-b060-72b28074b750) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/306065/jobs/0197446e-293b-487c-b060-72b28074b750/artifacts/0197447f-bef7-41c7-907c-9072b8cb9911) FTR Configs #<span></span>14 / Cases cases list bulk actions status change the status of cases to in-progress correctly
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/306065#0197446e-2971-41ba-9442-48557ebb66ab) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/306065/jobs/0197446e-2971-41ba-9442-48557ebb66ab/artifacts/01974479-1e2e-4eb2-912e-d5b5e30e1aab) FTR Configs #<span></span>128 / cases security and spaces enabled: trial Common update_alert_status should update the status of multiple alerts attached to multiple cases using the cases client
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/306065#01974479-34ec-4927-a198-0ce7d4849d0a) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/306065/jobs/01974479-34ec-4927-a198-0ce7d4849d0a/artifacts/0197447f-bccf-4456-abea-3bfd906a0763) FTR Configs #<span></span>128 / cases security and spaces enabled: trial Common update_alert_status should update the status of multiple alerts attached to multiple cases using the cases client

### History
* :broken_heart: [Build #306060](https://buildkite.com/elastic/kibana-pull-request/builds/306060) failed 567e15e2c173c5c81415d3c93115f5fd4c5fae02
* :broken_heart: [Build #306043](https://buildkite.com/elastic/kibana-pull-request/builds/306043) failed 174e27d8fdb2e108f4b2529a88650f8d10cecceb

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306065","number":306065,"commit":"9e72e8ca566e03e0a53c85f37fec4561319f0902"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306060","number":306060,"commit":"567e15e2c173c5c81415d3c93115f5fd4c5fae02"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306043","number":306043,"commit":"174e27d8fdb2e108f4b2529a88650f8d10cecceb"}],"number":306065}
buildkite-pr-comment-->

janmonschke committed:
fix unit tests

janmonschke committed:
disable id incrementer task on post case temporarily to investigate impact on FTR tests

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/306190)
* Commit: 5c0263b14666a98944a2f013d7cf03a554df7bd9

### Failed CI Steps
* [FTR Configs #1](https://buildkite.com/elastic/kibana-pull-request/builds/306190#01974542-1224-4c60-8afc-cb9ff909ff05)
* [FTR Configs #13](https://buildkite.com/elastic/kibana-pull-request/builds/306190#01974542-122e-4fc3-b3f6-b3476212d211)
* [FTR Configs #57](https://buildkite.com/elastic/kibana-pull-request/builds/306190#01974542-124f-4b64-b665-634608dc3249)
* [FTR Configs #124](https://buildkite.com/elastic/kibana-pull-request/builds/306190#01974542-1285-41b2-9202-b71a9394c8e9)
* [FTR Configs #124](https://buildkite.com/elastic/kibana-pull-request/builds/306190#01974559-1547-47f9-87c5-048ed8ebcbc4)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/306190#01974542-124f-4b64-b665-634608dc3249) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/306190/jobs/01974542-124f-4b64-b665-634608dc3249/artifacts/0197455f-c0a9-4447-9bbd-ddfef6040dc6) FTR Configs #<span></span>57 / Cases - group 1 View case properties status changes a case status to in-progress via dropdown menu
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/306190#01974542-1224-4c60-8afc-cb9ff909ff05) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/306190/jobs/01974542-1224-4c60-8afc-cb9ff909ff05/artifacts/01974560-7437-4f6f-949e-341897ae6a5a) FTR Configs #<span></span>1 / Cases cases list bulk actions status change the status of cases to in-progress correctly
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/306190#01974542-122e-4fc3-b3f6-b3476212d211) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/306190/jobs/01974542-122e-4fc3-b3f6-b3476212d211/artifacts/01974550-8f65-4b25-b7fd-007dc757b949) FTR Configs #<span></span>13 / cases security and spaces enabled: basic Common patch_comment partial updates should not result to a version conflict (409) when updating a comment to an updated case
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/306190#01974542-1285-41b2-9202-b71a9394c8e9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/306190/jobs/01974542-1285-41b2-9202-b71a9394c8e9/artifacts/01974559-0553-481f-947f-5582aa4f7114) FTR Configs #<span></span>124 / cases security and spaces enabled: trial Common find_cases basic tests correctly counts open/closed/in-progress
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/306190#01974559-1547-47f9-87c5-048ed8ebcbc4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/306190/jobs/01974559-1547-47f9-87c5-048ed8ebcbc4/artifacts/01974563-913b-46cd-aec8-7a2b836b7b09) FTR Configs #<span></span>124 / cases security and spaces enabled: trial Common patch_comment partial updates should not result to a version conflict (409) when updating a comment to an updated case

### History
* :broken_heart: [Build #306065](https://buildkite.com/elastic/kibana-pull-request/builds/306065) failed 9e72e8ca566e03e0a53c85f37fec4561319f0902
* :broken_heart: [Build #306060](https://buildkite.com/elastic/kibana-pull-request/builds/306060) failed 567e15e2c173c5c81415d3c93115f5fd4c5fae02
* :broken_heart: [Build #306043](https://buildkite.com/elastic/kibana-pull-request/builds/306043) failed 174e27d8fdb2e108f4b2529a88650f8d10cecceb

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306190","number":306190,"commit":"5c0263b14666a98944a2f013d7cf03a554df7bd9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306065","number":306065,"commit":"9e72e8ca566e03e0a53c85f37fec4561319f0902"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306060","number":306060,"commit":"567e15e2c173c5c81415d3c93115f5fd4c5fae02"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306043","number":306043,"commit":"174e27d8fdb2e108f4b2529a88650f8d10cecceb"}],"number":306190}
buildkite-pr-comment-->

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/common/types/id_incrementer.ts:
```diff
@@ -0,0 +1,21 @@
+
+export interface CaseIdIncrementerPersistedAttributes {
+  '@timestamp': number;
+  last_id: number;
```
I'm back and forth on the best intuitive name for this. Should it be `last_id` for the last id applied or should we be obvious like `next_id` or even `next_id_to_be_applied` or something. Last ID though does make the syncing with the last ID applied when we make that check easier... 🤔

michaelolo24 replied:
Not urgent enough to change, plus the saved object definitions were already merged here https://github.com/elastic/kibana/pull/219757. Thanks for the recommendations!

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/plugin.ts:
```diff
@@ -252,6 +262,9 @@ export class CasePlugin
             savedObjectsService: savedObjects,
           });
         },
+        scheduleIdCrementerTask: () => {
```
```suggestion
        scheduleIncrementerTask: () => {
```
or `scheduleIdIncrementerTask`?

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/services/incremental_id/index.test.ts:
```diff
@@ -0,0 +1,294 @@
+      expect(result).toStrictEqual(incIdSo);
+    });
+
+    it('should increase and persist `last_id` in case the last applied ID to a case is higher than in the inc ID SO', async () => {
```
👌🏾

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/services/incremental_id/index.ts:
```diff
@@ -0,0 +1,391 @@
+        return 0;
+      }
+
+      // TODO: should we really throw here?
```
I don't think we should throw here, but this could be a good telemetry point for how often this takes place, but still thinking through the best structure for the telemetry in general for this service

janmonschke replied:
This comment was outdated actually ffdb664e0e4

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/services/incremental_id/index.ts:
```diff
@@ -0,0 +1,391 @@
+  /**
+   * Increments the case ids for the given cases.
+   * @param casesWithoutIncrementalId The cases we want to apply IDs to
+   * @param maxDurationMs A maximum timeout to run this task for
```
How does this line up with the task manager?

janmonschke replied:
This piece of code is not needed anymore, I'll clean that up. I put it there before I learned that the task will be cancelled at intervals as well

janmonschke replied:
see [c267ece](https://github.com/elastic/kibana/pull/222874/commits/c267ece31201ba5b665f0b00fa1a1bf438be5600)

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/services/incremental_id/index.ts:
```diff
@@ -0,0 +1,391 @@
+
+      const actualLatestId = latestAppliedId || 0;
+
+      const incrementerSO = incrementerResponse?.saved_objects[0];
```
Though unlikely, what happens if the incrementer SO doesn't exist for whatever reason?

janmonschke replied:
Then the `else` case kicks in:

```js
// At this point we assume that no incrementer SO exists
this.logger.debug(`No incrementer found for ${namespace}. Creating a new one.`);
return this.createCaseIdIncrementerSo(namespace);
```

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/services/incremental_id/index.ts:
```diff
@@ -0,0 +1,391 @@
+    namespace: string
+  ) {
+    // Find the incrementer with the highest ID
+    const incrementerWithHighestId = incrementerQueryResponse.reduce<
```
nit, it may be a bit more performant to just do a single loop, and track these in two variables? I do prefer more implicit code, but I think overall performance is important for all the methods here in any worst case scenarios

`let incrementerWithHighestId = incrementerQueryResponse[0]`
`let incrementersToDelete = []; // Every time a larger incrementer is found, push the smaller one into the array`

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/tasks/incremental_id/incremental_id_task_manager.ts:
```diff
@@ -0,0 +1,143 @@
+        await this.taskManager?.runSoon(CASES_INCREMENTAL_ID_SYNC_TASK_ID);
+      }
+    } catch (e) {
+      this.logger.debug(`Could not run task: ${e}`);
```
Can we update this to
```suggestion
      this.logger.debug(`Could not run case incremental id task: ${e}`);
```

janmonschke commented on x-pack/platform/plugins/shared/cases/server/services/incremental_id/index.ts:
```diff
@@ -0,0 +1,391 @@
+
+  public async getCases({
+    filter,
+    perPage = 10000,
```
Good call, we should probably do 1k at time

janmonschke committed:
Update x-pack/platform/plugins/shared/cases/server/tasks/incremental_id/incremental_id_task_manager.ts

Co-authored-by: Michael Olorunnisola <michael.olorunnisola@elastic.co>

janmonschke committed:
rename scheduling function

janmonschke committed:
test stopped service behaviour

janmonschke committed:
only process 1k objects at a time

10k objects is too much to handle in a task and will definitely timeout

janmonschke committed:
remove custom timeout logic

The task's/service's timeout functionality will take care of it

janmonschke committed:
simplify duplicate incrementer resolution

janmonschke committed:
add ui setting for displaying the incremenetal id

janmonschke committed:
update interval

janmonschke committed:
update timeout

kibanamachine commented:
<!-- kbn-message-context:cloud-deploy-job -->
PR Cloud deployment started at: https://buildkite.com/elastic/kibana-deploy-cloud-from-pr/builds/237

kibanamachine commented:
<!-- kbn-message-context:cloud-deploy-job -->
Cloud deployment initiated, see credentials at: https://buildkite.com/elastic/kibana-deploy-cloud-from-pr/builds/237

janmonschke committed:
add incrementalIdService.enabled setting (enabled by default)

janmonschke committed:
disable incremental id service in cases FTR tests

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/yarn_deduplicate'

janmonschke committed:
fix i18n key

janmonschke committed:
fix types

janmonschke committed:
disable incremental id service in more tests

janmonschke committed:
Merge branch 'cases/incremental-ids-service-and-ui' of https://github.com/janmonschke/kibana into cases/incremental-ids-service-and-ui

janmonschke committed:
fix ftr config

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307882)
* Commit: c3ac5cf775fa3b4248279d80de36d9dc1eaf8c53
* [Cloud Deployment](https://kibana-pr-222874.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #307841](https://buildkite.com/elastic/kibana-pull-request/builds/307841) failed 6a6ab79f8087bb28ee63cc425656bf427e146165
* :broken_heart: [Build #307812](https://buildkite.com/elastic/kibana-pull-request/builds/307812) failed 85e09b5a5e44e83fc53602208cfe21a1830932a9
* :broken_heart: [Build #306190](https://buildkite.com/elastic/kibana-pull-request/builds/306190) failed 5c0263b14666a98944a2f013d7cf03a554df7bd9
* :broken_heart: [Build #306065](https://buildkite.com/elastic/kibana-pull-request/builds/306065) failed 9e72e8ca566e03e0a53c85f37fec4561319f0902
* :broken_heart: [Build #306060](https://buildkite.com/elastic/kibana-pull-request/builds/306060) failed 567e15e2c173c5c81415d3c93115f5fd4c5fae02

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307882","number":307882,"commit":"c3ac5cf775fa3b4248279d80de36d9dc1eaf8c53"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307841","number":307841,"commit":"6a6ab79f8087bb28ee63cc425656bf427e146165"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307812","number":307812,"commit":"85e09b5a5e44e83fc53602208cfe21a1830932a9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306190","number":306190,"commit":"5c0263b14666a98944a2f013d7cf03a554df7bd9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306065","number":306065,"commit":"9e72e8ca566e03e0a53c85f37fec4561319f0902"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306060","number":306060,"commit":"567e15e2c173c5c81415d3c93115f5fd4c5fae02"}],"number":307882}
buildkite-pr-comment-->

janmonschke committed:
fix tests

janmonschke committed:
fix search by incremental_id

janmonschke committed:
UX improvement for incremental id text

user-select:all will make sure that the # is always copied as well

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/styled_components_mapping'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308196)
* Commit: 6ed9997aee7ee8eb0ced886eec090456aa8e0ef1

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/308196#0197686c-58c3-499b-9b58-710b8424d747)

### History
* :broken_heart: [Build #307882](https://buildkite.com/elastic/kibana-pull-request/builds/307882) failed c3ac5cf775fa3b4248279d80de36d9dc1eaf8c53
* :broken_heart: [Build #307841](https://buildkite.com/elastic/kibana-pull-request/builds/307841) failed 6a6ab79f8087bb28ee63cc425656bf427e146165
* :broken_heart: [Build #307812](https://buildkite.com/elastic/kibana-pull-request/builds/307812) failed 85e09b5a5e44e83fc53602208cfe21a1830932a9
* :broken_heart: [Build #306190](https://buildkite.com/elastic/kibana-pull-request/builds/306190) failed 5c0263b14666a98944a2f013d7cf03a554df7bd9
* :broken_heart: [Build #306065](https://buildkite.com/elastic/kibana-pull-request/builds/306065) failed 9e72e8ca566e03e0a53c85f37fec4561319f0902

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308196","number":308196,"commit":"6ed9997aee7ee8eb0ced886eec090456aa8e0ef1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307882","number":307882,"commit":"c3ac5cf775fa3b4248279d80de36d9dc1eaf8c53"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307841","number":307841,"commit":"6a6ab79f8087bb28ee63cc425656bf427e146165"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307812","number":307812,"commit":"85e09b5a5e44e83fc53602208cfe21a1830932a9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306190","number":306190,"commit":"5c0263b14666a98944a2f013d7cf03a554df7bd9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306065","number":306065,"commit":"9e72e8ca566e03e0a53c85f37fec4561319f0902"}],"number":308196}
buildkite-pr-comment-->

janmonschke committed:
fix tests

janmonschke committed:
disable service in integration tests

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308316)
* Commit: 8b19ac18b1d99e59a19b98df881b21ffcb39ebea
* [Cloud Deployment](https://kibana-pr-222874.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #308206](https://buildkite.com/elastic/kibana-pull-request/builds/308206) failed 799a8549910cec2772678a0dba7e8b2d987d559f
* :broken_heart: [Build #308196](https://buildkite.com/elastic/kibana-pull-request/builds/308196) failed 6ed9997aee7ee8eb0ced886eec090456aa8e0ef1
* :broken_heart: [Build #307882](https://buildkite.com/elastic/kibana-pull-request/builds/307882) failed c3ac5cf775fa3b4248279d80de36d9dc1eaf8c53
* :broken_heart: [Build #307841](https://buildkite.com/elastic/kibana-pull-request/builds/307841) failed 6a6ab79f8087bb28ee63cc425656bf427e146165
* :broken_heart: [Build #307812](https://buildkite.com/elastic/kibana-pull-request/builds/307812) failed 85e09b5a5e44e83fc53602208cfe21a1830932a9

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308316","number":308316,"commit":"8b19ac18b1d99e59a19b98df881b21ffcb39ebea"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308206","number":308206,"commit":"799a8549910cec2772678a0dba7e8b2d987d559f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308196","number":308196,"commit":"6ed9997aee7ee8eb0ced886eec090456aa8e0ef1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307882","number":307882,"commit":"c3ac5cf775fa3b4248279d80de36d9dc1eaf8c53"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307841","number":307841,"commit":"6a6ab79f8087bb28ee63cc425656bf427e146165"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307812","number":307812,"commit":"85e09b5a5e44e83fc53602208cfe21a1830932a9"}],"number":308316}
buildkite-pr-comment-->

janmonschke committed:
remove outdated comment

janmonschke committed:
try catch

janmonschke commented on x-pack/platform/plugins/shared/cases/server/services/incremental_id/index.ts:
```diff
@@ -0,0 +1,391 @@
+    const incrementersToDelete = incrementerQueryResponse.filter(
+      (incrementer) => incrementer !== incrementerWithHighestId
+    );
+    await this.internalSavedObjectsClient.bulkDelete(incrementersToDelete);
```
Good call. It will get cleaned up in the next iteration anyways. 2e6fad436f6

janmonschke committed:
increase timeout

elasticmachine commented:
Pinging @elastic/response-ops-cases (Feature:Cases)

janmonschke committed:
Merge branch 'main' into cases/incremental-ids-service-and-ui

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308887)
* Commit: ab8ca99eec2fd2052eadc1600cd5e8c7b71467a8
* [Cloud Deployment](https://kibana-pr-222874.kb.us-west2.gcp.elastic-cloud.com)

### History
* :yellow_heart: [Build #308316](https://buildkite.com/elastic/kibana-pull-request/builds/308316) was flaky 8b19ac18b1d99e59a19b98df881b21ffcb39ebea
* :broken_heart: [Build #308206](https://buildkite.com/elastic/kibana-pull-request/builds/308206) failed 799a8549910cec2772678a0dba7e8b2d987d559f
* :broken_heart: [Build #308196](https://buildkite.com/elastic/kibana-pull-request/builds/308196) failed 6ed9997aee7ee8eb0ced886eec090456aa8e0ef1
* :broken_heart: [Build #307882](https://buildkite.com/elastic/kibana-pull-request/builds/307882) failed c3ac5cf775fa3b4248279d80de36d9dc1eaf8c53
* :broken_heart: [Build #307841](https://buildkite.com/elastic/kibana-pull-request/builds/307841) failed 6a6ab79f8087bb28ee63cc425656bf427e146165

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308887","number":308887,"commit":"ab8ca99eec2fd2052eadc1600cd5e8c7b71467a8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308316","number":308316,"commit":"8b19ac18b1d99e59a19b98df881b21ffcb39ebea"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308206","number":308206,"commit":"799a8549910cec2772678a0dba7e8b2d987d559f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308196","number":308196,"commit":"6ed9997aee7ee8eb0ced886eec090456aa8e0ef1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307882","number":307882,"commit":"c3ac5cf775fa3b4248279d80de36d9dc1eaf8c53"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307841","number":307841,"commit":"6a6ab79f8087bb28ee63cc425656bf427e146165"}],"number":308887}
buildkite-pr-comment-->

janmonschke commented on x-pack/test_serverless/shared/config.base.ts:
```diff
@@ -183,6 +183,7 @@ export default async () => {
         `--xpack.cloud.users_and_roles_url=/account/members/`,
         // disable product intercept for all ftr tests by default
         '--xpack.intercepts.enabled=false',
+        '--xpack.cases.incrementalIdService.enabled=false',
```
@pheyos The service is disabled in these tests because it results in flaky tests due to 409 errors in cases tests. The service processes cases asynchronously and in tests we often create then change cases immediately, faster than a human would, which results in conflicts.

janmonschke replied:
@pheyos Gotcha, I see your point now that I learned that MKI settings are only local. Thanks for raising this issue!

> Did you check what it would take to stabilize the tests for this new behavior?

Since the service acts completely asynchronous it is unclear to me what's the full list of tests that will fail. I'll re-enable the service in tests locally to see what I can do.

janmonschke replied:
@pheyos The tests have been enabled again. The task execution is delayed now and should not interfere with tests that are written in a way that don't account for async changes in between.

janmonschke replied:
@pheyos The tests have been enabled again. The task execution is delayed now and should not interfere with tests that are written in a way that don't account for async changes in between.

pheyos replied:
That's great @janmonschke! :tada: 
With that change, there's actually no `appex-qa` review required anymore for this PR.
I'd suggest to give the affected tests a go on the flaky test runner to ensure stability before merging this PR.

pheyos replied:
Awesome :tada:

janmonschke committed:
fast interval/timeout

janmonschke committed:
try to reenable some ftr tests

janmonschke committed:
remove implicit date-fns dependency

janmonschke committed:
remove flag from more configs

janmonschke committed:
remove empty argument

janmonschke committed:
Merge branch 'main' into cases/incremental-ids-service-and-ui

janmonschke committed:
use new config values

kibanamachine commented:
## Flaky Test Runner Stats
### 🎉 All tests passed! - [kibana-flaky-test-suite-runner#8415](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8415)
[✅] x-pack/test/cases_api_integration/security_and_spaces/config_trial.ts: 25/25 tests passed.
[✅] x-pack/test/api_integration/apis/cases/config.ts: 25/25 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F222874%2Fhead)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/310284)
* Commit: 01bd6b666b80550e681155b31a2e606221c3fb7a

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/310284#01978726-19f6-4490-925f-2701f3d0fdfb)

### History
* :yellow_heart: [Build #309719](https://buildkite.com/elastic/kibana-pull-request/builds/309719) was flaky 777bf31c32ac40479ccbeedddeafdc6701104650
* :broken_heart: [Build #309555](https://buildkite.com/elastic/kibana-pull-request/builds/309555) failed 3be625ce9f210f019030b3848a2e1b1e73d2db0f
* :green_heart: [Build #309472](https://buildkite.com/elastic/kibana-pull-request/builds/309472) succeeded 023e7bcb0b9934000991c2047f08ecc0b72abb81
* :broken_heart: [Build #309447](https://buildkite.com/elastic/kibana-pull-request/builds/309447) failed e9d2f17e089b31cc47281b4ca6746194004970ee
* :green_heart: [Build #308887](https://buildkite.com/elastic/kibana-pull-request/builds/308887) succeeded ab8ca99eec2fd2052eadc1600cd5e8c7b71467a8

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310284","number":310284,"commit":"01bd6b666b80550e681155b31a2e606221c3fb7a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309719","number":309719,"commit":"777bf31c32ac40479ccbeedddeafdc6701104650"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309555","number":309555,"commit":"3be625ce9f210f019030b3848a2e1b1e73d2db0f"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309472","number":309472,"commit":"023e7bcb0b9934000991c2047f08ecc0b72abb81"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309447","number":309447,"commit":"e9d2f17e089b31cc47281b4ca6746194004970ee"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308887","number":308887,"commit":"ab8ca99eec2fd2052eadc1600cd5e8c7b71467a8"}],"number":310284}
buildkite-pr-comment-->

janmonschke committed:
fix types

janmonschke commented:
> There are deployments with more than 1M cases, so the timeout limit will be hit. What will happen in this case? On the next run, will it pick up the remaining ones? Should we increase the limit to 30 minutes?

The task will only ever process 1k cases at a time. When it's done with that or if it gets cancelled by the timeout, it'll restart where it stopped last time. It was advised against increasing this timeout since a Kibana task slot is used for it and there are other tasks that need to process. I also vaguely remember there being a hard limit of 15 minutes or sth.

> Is this edge case acceptable by product? Does it happen to newly created cases? It would be interesting to see why it happened. It shouldn't.

It shouldn't skip any ids ideally and I'm not sure yet, how this can happen. Just to put this into context: All other solutions that were discussed had potential for gaps much larger than this (gaps of hundreds/thousands). So in the grand scheme of things, 8/50k is very little.

janmonschke committed:
fix tests

janmonschke committed:
Merge branch 'main' into cases/incremental-ids-service-and-ui

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/310470)
* Commit: e9aa06dc313ebec111e43c4e8181479725b57854
* [Cloud Deployment](https://kibana-pr-222874.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #310296](https://buildkite.com/elastic/kibana-pull-request/builds/310296) failed d83223fdd243d6ad099c4780b18354e907bf23d9
* :broken_heart: [Build #310284](https://buildkite.com/elastic/kibana-pull-request/builds/310284) failed 01bd6b666b80550e681155b31a2e606221c3fb7a
* :yellow_heart: [Build #309719](https://buildkite.com/elastic/kibana-pull-request/builds/309719) was flaky 777bf31c32ac40479ccbeedddeafdc6701104650
* :broken_heart: [Build #309555](https://buildkite.com/elastic/kibana-pull-request/builds/309555) failed 3be625ce9f210f019030b3848a2e1b1e73d2db0f
* :green_heart: [Build #309472](https://buildkite.com/elastic/kibana-pull-request/builds/309472) succeeded 023e7bcb0b9934000991c2047f08ecc0b72abb81

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310470","number":310470,"commit":"e9aa06dc313ebec111e43c4e8181479725b57854"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310296","number":310296,"commit":"d83223fdd243d6ad099c4780b18354e907bf23d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310284","number":310284,"commit":"01bd6b666b80550e681155b31a2e606221c3fb7a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309719","number":309719,"commit":"777bf31c32ac40479ccbeedddeafdc6701104650"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309555","number":309555,"commit":"3be625ce9f210f019030b3848a2e1b1e73d2db0f"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309472","number":309472,"commit":"023e7bcb0b9934000991c2047f08ecc0b72abb81"}],"number":310470}
buildkite-pr-comment-->

janmonschke committed:
remove unused scheduleIdIncrementerTask

janmonschke committed:
prevent double-schedules of task

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/310792)
* Commit: a5493d7312e28f84fff04d1acbbe644eaf967b10
* [Cloud Deployment](https://kibana-pr-222874.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #310706](https://buildkite.com/elastic/kibana-pull-request/builds/310706) succeeded 57516d329e0faf6b8fa3cd0788e589e7a18faa6f
* :yellow_heart: [Build #310470](https://buildkite.com/elastic/kibana-pull-request/builds/310470) was flaky e9aa06dc313ebec111e43c4e8181479725b57854
* :broken_heart: [Build #310296](https://buildkite.com/elastic/kibana-pull-request/builds/310296) failed d83223fdd243d6ad099c4780b18354e907bf23d9
* :broken_heart: [Build #310284](https://buildkite.com/elastic/kibana-pull-request/builds/310284) failed 01bd6b666b80550e681155b31a2e606221c3fb7a
* :yellow_heart: [Build #309719](https://buildkite.com/elastic/kibana-pull-request/builds/309719) was flaky 777bf31c32ac40479ccbeedddeafdc6701104650

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310792","number":310792,"commit":"a5493d7312e28f84fff04d1acbbe644eaf967b10"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310706","number":310706,"commit":"57516d329e0faf6b8fa3cd0788e589e7a18faa6f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310470","number":310470,"commit":"e9aa06dc313ebec111e43c4e8181479725b57854"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310296","number":310296,"commit":"d83223fdd243d6ad099c4780b18354e907bf23d9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310284","number":310284,"commit":"01bd6b666b80550e681155b31a2e606221c3fb7a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309719","number":309719,"commit":"777bf31c32ac40479ccbeedddeafdc6701104650"}],"number":310792}
buildkite-pr-comment-->

janmonschke committed:
move creation of objects around

cnasikas commented:
> The task will only ever process 1k cases at a time. When it's done with that or if it gets cancelled by the timeout, it'll restart where it stopped last time. It was advised against increasing this timeout since a Kibana task slot is used for it and there are other tasks that need to process. I also vaguely remember there being a hard limit of 15 minutes or sth.
>

Ok, but 1K seems very little to be able to timeout. When I reach the point in the code, I will understand better.

> It shouldn't skip any ids ideally and I'm not sure yet, how this can happen. Just to put this into context: All other solutions that were discussed had potential for gaps much larger than this (gaps of hundreds/thousands). So in the grand scheme of things, 8/50k is very little.

Maybe I misunderstood. Are there cases without an assigned incremental ID, or is there a gap between the numbers of the incremental ID (case A has #3 but case B has #5)?

janmonschke commented:
> Maybe I misunderstood. Are there cases without an assigned incremental ID, or is there a gap between the numbers of the incremental ID (case A has `#3` but case B has `#5`)?

The latter. A gap between numbers, yes. Cases that don't have an incremental id will always get picked up by the next iteration.

> Ok, but 1K seems very little to be able to timeout

Happy to increase it, it does not really matter too much tbh.

cnasikas reviewed (commented):
Hey! First pass of the review. I will continue with the review next week.

cnasikas commented:
> The latter. A gap between numbers, yes. Cases that don't have an incremental id will always get picked up by the next iteration.

Ok, that's fine then! I thought there were cases without IDs.

janmonschke committed:
remove singleton of `CasesIncrementalIdService`

In async scenarios, this can lead to side-effects due to shared state of `isStopped`

janmonschke committed:
add better test for async call of `stopService`

janmonschke committed:
fix test

janmonschke committed:
add reset logic

more context: https://github.com/elastic/kibana/pull/222874#issuecomment-2992584119

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311111)
* Commit: 00667d55f9f2551bd8314905c653baa84e509c6a
* [Cloud Deployment](https://kibana-pr-222874.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #311091](https://buildkite.com/elastic/kibana-pull-request/builds/311091) succeeded 49b96d8696032c09c592f3af51f08ff03cf0ac92
* :yellow_heart: [Build #310866](https://buildkite.com/elastic/kibana-pull-request/builds/310866) was flaky 82a411932f31247ede1c5b5a5e2b5e7c86bcff0c
* :yellow_heart: [Build #310792](https://buildkite.com/elastic/kibana-pull-request/builds/310792) was flaky a5493d7312e28f84fff04d1acbbe644eaf967b10
* :green_heart: [Build #310706](https://buildkite.com/elastic/kibana-pull-request/builds/310706) succeeded 57516d329e0faf6b8fa3cd0788e589e7a18faa6f
* :yellow_heart: [Build #310470](https://buildkite.com/elastic/kibana-pull-request/builds/310470) was flaky e9aa06dc313ebec111e43c4e8181479725b57854
* :broken_heart: [Build #310296](https://buildkite.com/elastic/kibana-pull-request/builds/310296) failed d83223fdd243d6ad099c4780b18354e907bf23d9

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311111","number":311111,"commit":"00667d55f9f2551bd8314905c653baa84e509c6a"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311091","number":311091,"commit":"49b96d8696032c09c592f3af51f08ff03cf0ac92"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310866","number":310866,"commit":"82a411932f31247ede1c5b5a5e2b5e7c86bcff0c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310792","number":310792,"commit":"a5493d7312e28f84fff04d1acbbe644eaf967b10"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310706","number":310706,"commit":"57516d329e0faf6b8fa3cd0788e589e7a18faa6f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310470","number":310470,"commit":"e9aa06dc313ebec111e43c4e8181479725b57854"}],"number":311111}
buildkite-pr-comment-->

afharo reviewed (approved):
Added UI settings LGTM

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/plugin.ts:
```diff
@@ -117,14 +121,39 @@ export class CasePlugin
+
+    if (plugins.taskManager) {
+      if (this.caseConfig.incrementalIdService.enabled) {
+        this.incrementalIdTaskManager = new IncrementalIdTaskManager(
```
Can modify to follow this pattern. The premise of our current pattern was established in the security solution codebase as seen here: https://github.com/elastic/kibana/blob/e2a833785b6f914aad49b1406b75fd4c9ad97bbc/x-pack/solutions/security/plugins/security_solution/server/plugin.ts#L466-L470 . There are some other examples in the security solution code base that do follow the pattern you recommend here so there isn't necessarily consistency there. If I have time tomorrow, I'll try to change the code base to match this pattern

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/tasks/incremental_id/incremental_id_task_manager.ts:
```diff
@@ -0,0 +1,151 @@
+              const initializedTime = new Date().toISOString();
+              const startTime = performance.now();
+              this.logger.debug(`Increment id task started at: ${initializedTime}`);
+              if (!this.casesIncrementService) {
```
Looks like @janmonschke already made this change

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/tasks/incremental_id/incremental_id_task_manager.ts:
```diff
@@ -0,0 +1,151 @@
+          },
+          params: {},
+          state: {},
+          scope: ['cases'],
```
https://github.com/elastic/kibana/blob/e2a833785b6f914aad49b1406b75fd4c9ad97bbc/x-pack/platform/plugins/shared/task_manager/server/task.ts#L382-L383

Just grouping and easier to query for a certain set of tasks. Could be useful to unify on this for the analytics index work as well 👍🏾

michaelolo24 committed:
Merge branch 'main' into cases/incremental-ids-service-and-ui

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311397)
* Commit: f3f84baf7ed7dd8a861ec1665eadcfe65f6a2847
* [Cloud Deployment](https://kibana-pr-222874.kb.us-west2.gcp.elastic-cloud.com)

### History
* :green_heart: [Build #311111](https://buildkite.com/elastic/kibana-pull-request/builds/311111) succeeded 00667d55f9f2551bd8314905c653baa84e509c6a
* :green_heart: [Build #311091](https://buildkite.com/elastic/kibana-pull-request/builds/311091) succeeded 49b96d8696032c09c592f3af51f08ff03cf0ac92
* :yellow_heart: [Build #310866](https://buildkite.com/elastic/kibana-pull-request/builds/310866) was flaky 82a411932f31247ede1c5b5a5e2b5e7c86bcff0c
* :yellow_heart: [Build #310792](https://buildkite.com/elastic/kibana-pull-request/builds/310792) was flaky a5493d7312e28f84fff04d1acbbe644eaf967b10
* :green_heart: [Build #310706](https://buildkite.com/elastic/kibana-pull-request/builds/310706) succeeded 57516d329e0faf6b8fa3cd0788e589e7a18faa6f

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311397","number":311397,"commit":"f3f84baf7ed7dd8a861ec1665eadcfe65f6a2847"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311111","number":311111,"commit":"00667d55f9f2551bd8314905c653baa84e509c6a"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311091","number":311091,"commit":"49b96d8696032c09c592f3af51f08ff03cf0ac92"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310866","number":310866,"commit":"82a411932f31247ede1c5b5a5e2b5e7c86bcff0c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310792","number":310792,"commit":"a5493d7312e28f84fff04d1acbbe644eaf967b10"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/310706","number":310706,"commit":"57516d329e0faf6b8fa3cd0788e589e7a18faa6f"}],"number":311397}
buildkite-pr-comment-->

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/tasks/incremental_id/incremental_id_task_manager.ts:
```diff
@@ -0,0 +1,151 @@
+        CASE_SAVED_OBJECT,
+        CASE_ID_INCREMENTER_SAVED_OBJECT,
+      ]);
+      const internalSavedObjectsClient = new SavedObjectsClient(internalSavedObjectsRepository);
```
We only want it scoped to the SO's we need for this task rather than having access to multiple saved objects unnecessarily

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/services/incremental_id/index.ts:
```diff
@@ -0,0 +1,395 @@
+    private internalSavedObjectsClient: SavedObjectsClientContract,
+    private logger: Logger
+  ) {
+    this.logger = logger.get('incremental_id_service');
```
Yes, it's the logger defined in the plugin.ts - this just scopes it further down for this service

michaelolo24 committed:
Merge remote-tracking branch 'upstream/main' into cases/incremental-ids-service-and-ui

michaelolo24 committed:
extend flag to uI

michaelolo24 committed:
rename incrementalIdService key to incrementalId, modularize uiSettings register

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/config.ts:
```diff
@@ -27,7 +27,7 @@ export const ConfigSchema = schema.object({
      * Whether the incremental id service should be enabled
      */
-    enabled: schema.boolean({ defaultValue: true }),
+    enabled: schema.boolean({ defaultValue: false }),
```
Feature will be default disabled for now

michaelolo24 reviewed (approved):
Extended the feature flag to the UI and all functionality will be default disabled until all additional testing is complete.

michaelolo24 committed:
Merge branch 'main' into cases/incremental-ids-service-and-ui

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/312341)
* Commit: 1fe971029b35d9268868b5cc6d9ac0fbe52aa5ba
* [Cloud Deployment](https://kibana-pr-222874.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #312289](https://buildkite.com/elastic/kibana-pull-request/builds/312289) failed 5629efde9d40a9eafd126f7207782df8c9422d75
* :green_heart: [Build #312206](https://buildkite.com/elastic/kibana-pull-request/builds/312206) succeeded de37ff5cf77c615fafd0ec2fbb1c51da03e7e0f6
* :green_heart: [Build #311397](https://buildkite.com/elastic/kibana-pull-request/builds/311397) succeeded f3f84baf7ed7dd8a861ec1665eadcfe65f6a2847
* :green_heart: [Build #311111](https://buildkite.com/elastic/kibana-pull-request/builds/311111) succeeded 00667d55f9f2551bd8314905c653baa84e509c6a
* :green_heart: [Build #311091](https://buildkite.com/elastic/kibana-pull-request/builds/311091) succeeded 49b96d8696032c09c592f3af51f08ff03cf0ac92

cc @janmonschke

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312341","number":312341,"commit":"1fe971029b35d9268868b5cc6d9ac0fbe52aa5ba"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312289","number":312289,"commit":"5629efde9d40a9eafd126f7207782df8c9422d75"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312206","number":312206,"commit":"de37ff5cf77c615fafd0ec2fbb1c51da03e7e0f6"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311397","number":311397,"commit":"f3f84baf7ed7dd8a861ec1665eadcfe65f6a2847"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311111","number":311111,"commit":"00667d55f9f2551bd8314905c653baa84e509c6a"},{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311091","number":311091,"commit":"49b96d8696032c09c592f3af51f08ff03cf0ac92"}],"number":312341}
buildkite-pr-comment-->

michaelolo24 committed:
fix tests

michaelolo24 commented on x-pack/platform/plugins/shared/cases/server/config.ts:
```diff
@@ -23,6 +23,26 @@ export const ConfigSchema = schema.object({
+    /**
+     * The interval that the task should be scheduled at
+     */
+    taskInterValMinutes: schema.number({
```
I've gone back and forth on this one, and while I agree, I think for some of our larger customers or customers wanting something closer to a "real time experience" may want to test shortening this task interval dramatically from the current defaults. 

Lastly, these configurations are generic enough to apply to the task itself and aren't concerned with the internals of the task. For instance, if we decided we wanted to parallelize the id application (probably never given the complexity), these would still apply.

azasypkin reviewed (approved):
Changes in `src/platform/test/plugin_functional/test_suites/core_plugins/rendering.ts` LGTM (exposed simple boolean flag).

kibanamachine commented:
Starting backport for target branches: 8.19

https://github.com/elastic/kibana/actions/runs/15875828143

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.19|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 222874
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
